test/unit/AssetUtils.test.ts

```diff
@@ -0,0 +1,112 @@
+import { assert } from 'chai'
+import { KNOWN_CONFIDENTIAL_EVMS } from '../../src/config'
+import { provider, getAddresses } from '../config'
+import {
+  calculateActiveTemplateIndex,
+  getTemplateIndexOnList,
+  isConfidentialEVM
+} from '../../src/utils'
+import { Signer } from 'ethers/lib/ethers'
+
+let nftOwner: Signer
+let addresses: any
+describe('Asset utils (createAsset)', () => {
+  before(async () => {
+    nftOwner = (await provider.getSigner(0)) as Signer
+    addresses = await getAddresses()
+  })
+
+  it('should check if confidential EVM', async () => {
+    for (const network of KNOWN_CONFIDENTIAL_EVMS.networks) {
+      network.name.map((networkName) => {
+        assert(
+          isConfidentialEVM(networkName) === true,
+          `Network: "${networkName}" is not a confidental EVM`
+        )
+        return true
+      })
+    }
+    for (const chain of KNOWN_CONFIDENTIAL_EVMS.networks) {
+      assert(
+        isConfidentialEVM(chain.chainId) === true,
+        `Chain Id: "${chain}" is not a confidental EVM`
+      )
+    }
+
+    // optimism sepolia
+    // 11155420
+    assert(
+      isConfidentialEVM(11155420) === false,
+      `Chain Id: "11155420" is wrongly considered a confidental EVM`
+    )
+  })
+
+  it('should get correct template index from contract artifacts (using SC address as template)', async () => {
```

Comment:
Why do we need this?
Never assume any hardcoded address.
Maybe next week we will change the templates, cause we have found a bug.  Or deploy a new product..
Then all ocean.js tests will fail, although they shoudn't